### ISSUE TITLE
Bug: `chainrules` einsum HVP examples are stale and corrected `Tape::hvp` runs return zero HVP

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#413]
## Summary

The public `chainrules` tensor AD examples are broken for a downstream user in two layers:

1. The crate docs and `docs/design/autodiff.md` still call `tracked_einsum` / `dual_einsum` without the now-required backend context arguments, so copy-paste usage fails at compile time.
2. After correcting those call sites to the live API, the advertised `Tape::hvp` path through `tracked_einsum` still returns the wrong result. For the documented `f(x) = x·x` example with `x = ones([3])` and tangent direction `v = ones([3])`, the expected HVP is `[2.0, 2.0, 2.0]`, but the external run returns `[0.0, 0.0, 0.0]`.

## Reproduction

Verified with two external consumer crates:

```bash
cargo run --offline --manifest-path target/tmp/bug-sweep-chainrules-external/Cargo.toml
cargo run --offline --manifest-path target/tmp/bug-sweep-chainrules-corrected/Cargo.toml
cargo test -p tenferro-einsum hvp_via_fw_grad_composition -- --nocapture
```

### Docs-exact consumer

Copy the public `chainrules` tensor examples essentially as written from:

- `extern/chainrules/src/lib.rs`
- `docs/design/autodiff.md`

Observed compile failure:

```text
error[E0061]: this function takes 3 arguments but 2 arguments were supplied
```

The stale call sites are things like:

```rust
let c = tracked_einsum("ij,jk->ik", &[&a, &b]).unwrap();
let c_dual = dual_einsum("ij,jk->ik", &[&a_dual, &b_dual]).unwrap();
let loss = tracked_einsum("i,i->", &[&x, &x]).unwrap();
```

### Corrected consumer

After correcting the calls to the live API shape:

- `tracked_einsum::<Standard<f64>, CpuBackend>(ctx.clone(), ...)`
- `dual_einsum::<Standard<f64>, CpuBackend>(&mut ctx, ...)`

reverse-mode pullback, forward-mode JVP, and the `Variable` APIs all work, but `Tape::hvp` through `tracked_einsum` still returns the wrong answer.

Observed output:

```text
custom_grad=1
reverse_ga=[24.0, 24.0, 24.0, 24.0, 24.0, 24.0]
forward_jvp=[2.0, 2.0, 2.0, 2.0]
hvp_grad=[2.0, 2.0, 2.0]
hvp_hvp=[0.0, 0.0, 0.0]
backward_grad=4
grad_tangent=[4.0]
grad_variable_value=4 requires_grad=true
backward_hvp_grad=4
backward_hvp_hvp=2
```

For the documented HVP example, `hvp_grad=[2,2,2]` is correct but `hvp_hvp=[0,0,0]` is not.

## Expected behavior

- Public `chainrules` tensor AD examples should compile as documented.
- The documented `Tape::hvp` einsum example should produce `H·v = 2v` for `f(x) = x·x`.

## Actual behavior

- Docs-exact users hit compile errors immediately because the examples omit required backend contexts.
- Corrected external `Tape::hvp` runs through `tracked_einsum` lose the tangent seed and return zero HVP.

## Suspected root cause

`tracked_einsum` records reverse-rule `input_tangents` from `TrackedTensor` primal `fw_grad`:

- `tenferro-einsum/src/ad.rs:238-241`

and `pullback_with_tangents` only propagates those recorded tangents:

- `tenferro-einsum/src/ad.rs:102-126`

That does not line up with the public `Tape::leaf_with_tangent` / `Tape::hvp` route from `chainrules`, where the tangent direction is stored on the tape leaf rather than on `Tensor::fw_grad`.

By contrast, the variable path records tangents from `Variable::tangent()`:

- `tenferro-einsum/src/ad.rs:321`

This matches the observed split:

- existing in-repo `fw_grad` composition test passes
- external `TrackedTensor` / `Tape::hvp` path returns zero HVP

## Source references

- `extern/chainrules/src/lib.rs:44`
- `extern/chainrules/src/lib.rs:45`
- `extern/chainrules/src/lib.rs:88`
- `extern/chainrules/src/lib.rs:105`
- `extern/chainrules/src/lib.rs:108`
- `docs/design/autodiff.md:236`
- `docs/design/autodiff.md:250`
- `tenferro-einsum/src/ad.rs:102`
- `tenferro-einsum/src/ad.rs:119`
- `tenferro-einsum/src/ad.rs:125`
- `tenferro-einsum/src/ad.rs:238`
- `tenferro-einsum/src/ad.rs:321`
- `tenferro-einsum/tests/einsum_tests.rs:2294`
- `tenferro-einsum/tests/einsum_tests.rs:2366`

## Report

- `docs/test-reports/agentic-bug-sweep/bug-sweep-20260309-071823.md`

## Related issues

- #408

## Related issues

- #408
